streams/src/test/java/org/apache/kafka/streams/processor/internals/assignment/StickyTaskAssignorTest.java

```diff
@@ -674,6 +674,19 @@ public void shouldAssignTasksToNewClientWithoutFlippingAssignmentBetweenExisting
         assertThat(newClient.activeTaskCount(), equalTo(2));
     }
 
+    @Test
+    public void shouldViolateBalanceToPreserveActiveTaskStickiness() {
```

Comment:
Does this test pass on the current code base, or is it a consequence of this PR?

Reply:
It's testing the newly added `#preservePreviousTaskAssignment`: this is sort of a temporary hack to force the `StickyTaskAssignor` to return a "completely sticky" active task assignment, in the case the admin `listOffsets` request fails or times out.

The default `StickyTaskAssignor` behavior is unchanged in this PR, to leave the code in a stable state and avoid adding a huge refactoring of `StickyTaskAssignorTest` to an already large PR.

Reply:
Ok, thanks. The subtlety was a bit lost on me.